File: fibonacci/fib.cpp
#include <vector>
#include <iostream>

using namespace std;

typedef vector<int> IntVec;

int fib(int n) {
    IntVec vals(n+1, 0);
    vals[1]=1;
    if(n < 2)
        return vals[n];
    for(int i=2; i<n+1; ++i) {
        if(!vals[i])
            vals[i]=vals[i-1]+vals[i-2];
    }
    return vals[n];
}
        
int main() {
    for(int i=2; i < 45; ++i)
        cout << i << ":" << fib(i) << endl;    
}

File: fibonacci/fib.py
#!/usr/bin/python3


def fib(n):
    vals=[0]*(n+1)
    vals[1]=1
    if(n < 2):
        return vals[n]
    for i in range(2, n+1): 
        vals[i]=vals[i-1]+vals[i-2]
    return vals[n]
        
def main(): 
    for i in range(2, 45):
        print("{}:{}".format(i,fib(i)))

if __name__=="__main__":
    main()

